### ISSUE TITLE
Consider upstream integration

### ISSUE THREAD

**Issue by dsieger** [jothepro/doxygen-awesome-css#30]
Hi again,

Did you consider submitting your work upstream to be integrated directly into Doxygen? While I really like your solution, I'm a little concerned about incompatibilities with future Doxygen versions. Furthermore, this would allow a much larger group of people to benefit from you work. Just a thought...

**Comment by jothepro** [comment]
Yep, that's a good idea. I am already in contact with the Maintainer of doxygen: https://github.com/doxygen/doxygen/issues/6658

Nevertheless I' prefer to give the theme some time to incubate in this repository, with a smaller user base. And I expect bringing the theme officially to Doxygen to be very labor intensive. As long as I can achieve what I want by patching the existing theme I guess my incentive to contribute to Doxygen will be too small for it to happen anytime soon. :/

**Comment by tdegeus** [comment]
I like you theme very much! It is probably a very good idea to mature the theme here before integrating upstream. Indeed fewer users, but also much faster release cycle. However, I too would love to see it integrated upstream. In the end that is  simpler as a user, but probably limits the number of work-arounds in the long run.

**Comment by tdegeus** [comment]
It seems that doxygen has released a dark theme : https://github.com/doxygen/doxygen/issues/8404 . I would feel that the time is right to integrate upstream. Since there is now a configuration option `HTML_COLORSTYLE` your style could potentially just be one of the options?

**Comment by jothepro** [comment]
I feel like I should give you an update about my current thoughts on this topic as a maintainer:

Atm I have to patch the theme for new releases of doxygen on a regular basis, which I realized is a risky strategy to go forward. There may at one point be changes that cannot be "fixed" with a few more lines of CSS overrides added to the theme. In fact this has [already happened](https://github.com/jothepro/doxygen-awesome-css/issues/86) with the doxygen 1.9.5 release.

I'll definitely have to evaluate strategies on how to mitigate this risk in the near future. 
My current plan is to work on a standalone version of the theme that does not depend on the default `doxygen.css`. This would decouple the styling from new doxygen releases. It is still risky though bc the HTML may also change over time.

At that point one may propose in [doxygen/doxygen](https://github.com/doxygen/doxygen) to add a switch to choose between the "default" and the "awesome" (🤭) theme variant. Thanks @tdegeus for the idea to make this a `HTML_COLORSTYLE` option!

Any initiative from my side to create the standalone version of the theme and discuss at [doxygen/doxygen](https://github.com/doxygen/doxygen) how an integration would be possible will likely not happen in the next couple of months. But working towards this goal seems inevitable for the future of this theme and it will be my main focus in the mid term.
